bittensor/utils/async_substrate.py

```diff
@@ -344,7 +349,8 @@ async def get_block_number(self, block_hash: str) -> int:
                 return int(response["result"]["number"], 16)
 
     def close(self):
-        raise NotImplementedError()
+        self.substrate.close()
+        # TODO: Websocket close need to be implemented and added here.
```

Comment:
Is there a reason we want to manually allow closing the connection?

Reply:
For the same reasons as in original SubstrateInterface. Then it used in Subtensor.